.github/workflows/ci-unit_tests.yaml

```diff
@@ -0,0 +1,61 @@
+name: CI Unit Tests
+on: [pull_request, workflow_dispatch]
+
+jobs:
+
+  ci_pytest:
+      runs-on: ubuntu-latest
+      name: Run unit tests on CI system      
+      permissions:
+        checks: write
+
+      steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.7.16
```

Comment:
Probably want to use 3.10+

Reply:
or not.

Reply:
I recall there was a specific reason I chose that particular version to three decimal places, but it escapes me at this present moment, but probably not material.